streaming_providers/debrid_client.py

```
@@ -1,121 +1,194 @@
         max_retries: int,
         retry_interval: int,
-    ):
+        torrent_info: Optional[dict] = None,
+    ) -> dict:
         """Wait for the torrent to reach a particular status."""
-        retries = 0
-        while retries < max_retries:
-            torrent_info = self.get_torrent_info(torrent_id)
+        # if torrent_info is available, check the status from it
+        if torrent_info:
             if torrent_info["status"] == target_status:
```

Ruff

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)